xbmc/guilib/Resolution.cpp

```diff
@@ -63,16 +67,87 @@ RESOLUTION CResolutionUtils::ChooseBestResolution(float fps, int width, bool is3
 {
   RESOLUTION res = g_graphicsContext.GetVideoResolution();
   float weight;
+
   if (!FindResolutionFromOverride(fps, width, is3D, res, weight, false)) //find a refreshrate from overrides
   {
     if (!FindResolutionFromOverride(fps, width, is3D, res, weight, true))//if that fails find it from a fallback
-      FindResolutionFromFpsMatch(fps, width, is3D, res, weight);//if that fails use automatic refreshrate selection
+    {
+      if (!FindResolutionFromWhitelist(fps, width, is3D, res)) //find a refreshrate from whitelist
+        FindResolutionFromFpsMatch(fps, width, is3D, res, weight);//if that fails use automatic refreshrate selection
```

Comment:
why is there a fallback to old behaviour? if a resulution is not whitelisted, it must not be selected

Reply:
You're right, I'll change that. I'll check if the whitelist is empty and return false. Otherwise if there is something in the whitelist it should always return true. 

Does that sound right?

Reply:
nope, a res must be whitelisted to get selected. 
- FindResolutionFromWhitelist does not need a return value.
- FindResolutionFromFpsMatch can be deleted

Reply:
No, I'm not going to force people to use the whitelist if they are happy with the current behaviour.

Reply:
there’s one compromise that comes into my mind:
if no whitelist is populated yet, add all resolutions to it, so everything would be whitelisted by default.
what do you think?

Reply:
>  I'm not going to force people to use the whitelist if they are happy with the current behaviour.

current brahavior will disappear after this change and allowing downscale

Reply:
that means: as soon we allow downscale, whitelist has to be used